### DISCUSSION TITLE
How to get the url hash tag from a server component using app/ folder?

### DISCUSSION THREAD

**Discussion by Franzx41** [vercel/next.js#62914] [Help] [answered]
### Summary

Assuming we have a url like:

`/page#ourtag`

There is any way to get the hash 'ourtag' from a server component using app/ folder?
I know that using server components from pages/ folder we could use getServerSideProps({ req, res }) and then get the hash from 'req' property but how we can do it using server components inside the app/ folder?

**Comment by mutasim77** [comment, marked as answer]
You cannot directly access the hash fragment `#anyhash` from the server component itself because the hash fragment is not sent to the server in HTTP requests. The hash fragment is only available on the client-side.

One way to achieve this is by utilizing query parameters: you can simply pass the hash fragment as a query parameter in the URL. Then, you can access it in your server components using `req.query`

Using `window.location.hash` will provide you with the URL hash. You can then append it to the URL as a query parameter, like 'url?hash=yourhash', and the server component can access it.

You can also refer to these answers: [link](https://stackoverflow.com/questions/317760/how-to-get-url-hash-from-server-side).

**Comment by icyJoseph** [comment]
As said above, the browser doesn't share the fragment/hash part of a URL with the server. I answered on a discussion here:

- https://github.com/vercel/next.js/discussions/56967#discussioncomment-7313944

**Comment by Franzx41** [comment]
Thank you guys for the clarification.
I ended up using the searchParams on my server component. Change my to

`/page?ourtag=data`
